### PR TITLE
Add support for PostgreSQL cast operator

### LINKED ISSUE #66: [Question] PostgreSQL Cast Operator
I noticed the PostgreSQL cast shortcut is incorrectly split, for example `123::bigint` becomes `123: :bigint`.

I can look at making a PR to support this, but first I wanted to check if there was interest in this tool supporting *platform specific* syntax... ?

### PR DESCRIPTION
**Pull request by forrest79**
Fix #66.

### CODE REVIEW

**Review by greg0ire — DISMISSED** [review]

File: src/Tokenizer.php [hunk 1]
```diff
@@ -857,6 +857,11 @@ private function createNextToken(string $string, ?Token $previous = null): Token
             );
         }
 
+        // PostgreSQL cast operator
+        if (substr($string, 0, 2) === '::') {
+            return new Token(Token::TOKEN_TYPE_BOUNDARY, '::');
+        }
```
Comment (greg0ire) [inline src/Tokenizer.php]: I think the same can be achieved by adding `::` to this list: https://github.com/doctrine/sql-formatter/blob/21139be16870171c4f067de6d0c700769a22b9a3/src/Tokenizer.php#L713

It has to be added before `:` though.

**Comment by greg0ire** [comment]
This should target 1.2.x as it isn't a bugfix.

File: src/Tokenizer.php (hunk 1 above)
Comment (forrest79) [reply src/Tokenizer.php]: You're right, it works :-)

**Review by greg0ire — APPROVED** [review]

File: tests/format.html [hunk 2]
```diff
@@ -973,3 +973,6 @@
     ORDER BY
       hire_date
   );
+---
+SELECT
+  1 :: text;
```
Comment (goetas) [inline tests/format.html:978]: TBH i do not like this formatting. most of the tools use `1::text` with no extra spaces.

**Review by goetas — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @forrest79 !

**Pull request merged**